packages/trpc/server/routers/viewer/filterSegments/delete.handler.ts

```diff
@@ -0,0 +1,84 @@
+import { prisma } from "@calcom/prisma";
+import type { TrpcSessionUser } from "@calcom/trpc/server/types";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TDeleteFilterSegmentInputSchema } from "./delete.schema";
+
+export const deleteHandler = async ({
+  ctx,
+  input,
+}: {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TDeleteFilterSegmentInputSchema;
+}) => {
+  const { id } = input;
+  const userId = ctx.user.id;
+
+  const segment = await prisma.filterSegment.findFirst({
+    where: {
+      id,
+    },
+    select: {
+      scope: true,
+      teamId: true,
+    },
+  });
+  if (!segment) {
+    throw new TRPCError({
+      code: "NOT_FOUND",
+      message: "Filter segment not found or you don't have permission to delete it",
+    });
+  }
```

Comment:
Im not sure we need this first query - we are searching by ID in the next query below. I think we can remove this and modify the latter query as follows